slsim/Deflectors/compound_lens_halos_galaxies.py

```diff
@@ -0,0 +1,146 @@
+        if "mag_i" not in column_names:
+            halo_galaxy_list["mag_i"] = -np.ones(n)
+        if "mag_z" not in column_names:
+            halo_galaxy_list["mag_z"] = -np.ones(n)
+        if "mag_Y" not in column_names:
+            halo_galaxy_list["mag_Y"] = -np.ones(n)
+        if "e1_light" not in column_names or "e2_light" not in column_names:
+            halo_galaxy_list["e1_light"] = -np.ones(n)
+            halo_galaxy_list["e2_light"] = -np.ones(n)
+        if "e1_mass" not in column_names or "e2_mass" not in column_names:
+            halo_galaxy_list["e1_mass"] = -np.ones(n)
+            halo_galaxy_list["e2_mass"] = -np.ones(n)
+
+        self._galaxy_select = deflector_cut(halo_galaxy_list, **kwargs_cut)
+        # Currently only supporting redshift cut
+        self._num_select = len(self._galaxy_select)
+
+        self._cosmo = cosmo
+
+        # TODO: random reshuffle of matched list
+
+    def deflector_number(self):
+        """
+
+        :return: number of deflectors
+        """
+        number = self._num_select
+        return number
+
+    def draw_deflector(self):
+        """
+        :return: dictionary of complete parameterization of deflector
+        """
+
+        cosmo = self._cosmo
+        index = random.randint(0, self._num_select - 1)
+        deflector = self._galaxy_select[index]
+        if deflector["vel_disp"] == -1:
+            theta_eff = deflector["tb"] / 0.551  # [arcsec]
+            reff = (
+                theta_eff
+                * cosmo.angular_diameter_distance(deflector["z"])
+                * constants.arcsec
+            ).value  # physical Mpc
+            vel_disp = vel_disp_composite_model(
+                theta_eff,
+                deflector["stellar_mass"],
+                reff,
+                max(deflector["halo_mass"], deflector["halo_mass_acc"]),
+                deflector["concentration"],
+                cosmo,
+                deflector["z"],
+            )
+            deflector["vel_disp"] = vel_disp
+        # if (
+        #     deflector["mag_g"] == -1
+        #     or deflector["mag_r"]
+        #     or deflector["mag_i"] == -1
+        #     or deflector["mag_z"]
+        #     or deflector["mag_Y"] == -1
+        # ):
+        #     mag_g, mag_r, mag_i, mag_z, mag_Y = (
+        #         0,
+        #         0,
+        #         0,
+        #         0,
+        #         0,
+        #     )  # TODO: make function if needed
+        if deflector["e1_light"] == -1 or deflector["e2_light"] == -1:
+            e1_light, e2_light, e1_mass, e2_mass = (
+                elliptical_projected_eccentricity_galaxy(**deflector)
+            )  # TODO: check
+            deflector["e1_light"] = e1_light
+            deflector["e2_light"] = e2_light
+            deflector["e1_mass"] = e1_mass
+            deflector["e2_mass"] = e2_mass
+        return deflector
+
+
+def elliptical_projected_eccentricity_galaxy(e_g, **kwargs):
```

Comment:
I made quite a few changes of this function here: https://github.com/LSST-strong-lensing/slsim/blob/6cffdf68434f343ab46f77534bf21db1e0a934cc/slsim/Deflectors/elliptical_lens_galaxies.py#L105 
Perhaps you will be able to use this one instead? Note my previous mistake of cos(2phi) instead of cos(phi)

Reply:
I changed to use the elliptical_projected_eccentricity function in slsim/slsim/Deflectors/elliptical_lens_galaxies.py when calculating the eccentricity of galaxies in compound_lens_halos_galaxies.py.